scripts/release/cli.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { exec } = require('child-process-promise');
+const { createPromptModule } = require('inquirer');
+const prompt = createPromptModule();
+const { hasUpdatedPackages } = require('./utils/lerna');
+const {
+  getAllPackages,
+  getOrderedUpdates,
+  mapPkgNameToPkgJson,
+  updateWorkspaceVersions
+} = require('./utils/workspace');
+const {
+  cleanTree,
+  commitAndTag,
+  getCurrentSha,
+  hasDiff,
+  pushUpdatesToGithub,
+  resetWorkingTree,
+  treeAtHead
+} = require('./utils/git');
+const {
+  packageVersionUpdate,
+  releaseType: releaseTypePrompt,
+  validateReadyToPush,
+  validateVersions
+} = require('./utils/inquirer');
+const { reinstallDeps } = require('./utils/yarn');
+const { runTests, setupTestDeps } = require('./utils/tests');
+const { publishToNpm } = require('./utils/npm');
+const { bannerText } = require('./utils/banner');
+const { argv } = require('yargs');
+
+(async () => {
+  try {
+    /**
+     * Welcome to the firebase release CLI!
+     */
+    await bannerText();
+
+    /**
+     * If there are unstaged changes, error
+     */
+    if (await hasDiff()) {
```

Comment:
Should probably check which branch this is on at this point.

Reply:
Good call.

Reply:
Updated the comment above, WDYT?